backend/tests/vehiclePropertyControllerTest.js

```diff
@@ -0,0 +1,151 @@
+let vehicle_prop_2_id = ''
+let vehicle_prop_1 = {}
+let vehicle_prop_2 = {}
+
+// The following tests test the entire Vehicle Property Controller Class
+describe('Testing Vehicle Property Controller Class', () => {
+  //The below test checks if we can create a vehicle property successfully
+  describe('Create Vehicle Property API Test', () => {
+    it('should return 200 when a vehicle property is created', (done) => {
+      chai.request(app)
+        .post("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties")
+        .send({
+            'dealership': '618b3bf134f07d9a91c32a1b',
+            'label': 'Test1',
+            'inputType': 'Text',
+            'visible': false
+        })
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          vehicle_prop_1_id = response.body.payload._id;
+          vehicle_prop_1 = response.body.payload;
+
+          done();
+        });
+    });
+  });
+
+  describe('Create 2nd Vehicle Property API Test', () => {
+    it('should return 200 when a vehicle property is created', (done) => {
+      chai.request(app)
+        .post("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties")
+        .send({
+            'dealership': '618b3bf134f07d9a91c32a1b',
+            'label': 'Test2',
+            'inputType': 'Text',
+            'visible': false,
+        })
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          vehicle_prop_2_id = response.body.payload._id;
+          vehicle_prop_2 = response.body.payload;
+          done();
+        });
+    });
+  });
+
+  //The below test checks if we can fetch a vehicle property successfully
+  describe('Get Vehicle Properties API Test', () => {
+    it('should return 200 when vehicle properties are returned', (done) => {
+      chai.request(app)
+        .get("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties")
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          done();
```

Comment:
Add an assert statement to check that the payload is a non-empty array. Makes for a more descriptive test.

Reply:
I have optimized the above test in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/34/commits/2c1fa39cd96d281067f13e5628c2e5edf104968d)